### ISSUE TITLE
/metrics endpoint is resolving but no metrics are displayed

### ISSUE THREAD

**Issue by mark-e-hoffman** [sd2k/rocket_prometheus#21]
I'm using version = "0.10.0-rc.3" and my rocket webserver will resolve /metrics but there is no data returned, even though I have sent several GET AND POST requests to my server.

rocket::build()
        .attach(AdHoc::config::<Config>())
        .manage(pool)
        .manage(replier)
        .manage(prometheus.clone())
        .manage(amqp_pool)
         .attach(rocket_governor::LimitHeaderGen::default())
        .register("/", catchers!(rocket_governor_catcher)) // catches limit errors and produces a 429 error code
        .mount("/ref", routes![refs::get,refs::patch, refs::delete, refs::post, refs::get_all])
        .mount("/device", routes![device::get,device::patch, device::delete, device::post, device::get_all])
        .mount("/library", routes![library::get,library::patch, library::delete, library::post, library::get_all, library::link_ref_item_to_library, library::get_ref_item_in_library])
        .mount("/session", routes![session::get,session::patch, session::delete, session::post, session::get_all])
        .mount("/project", routes![project::get,project::patch, project::delete, project::post, project::get_all,project::link_library_to_project, project::get_library_in_project,project::match_detections])
        .mount("/metrics", prometheus)
        .launch().await?;

**Comment by sd2k** [comment]
I can't see anything wrong with your setup in the code snippet you've posted, but it's difficult to reproduce such a complicated example. I've never noticed any issues like that in my apps. The only thing I can think of is to try changing the order of your mounts: perhaps the `.register("/", catchers!(rocket_governor_catcher)` should be after the `.mount()` calls?

**Comment by mark-e-hoffman** [comment]
thanks for responding so quickly.. Unfortunately, that didn't make any
difference.
I just get this in the rocket log

2023-06-22T19:12:20.673Z INFO  [rocket::server] GET /metrics text/html:
2023-06-22T19:12:20.673Z INFO  [rocket::server::_] Matched: GET /metrics/
2023-06-22T19:12:20.673Z INFO  [rocket::server::_] Outcome: Success
2023-06-22T19:12:20.673Z INFO  [rocket::server::_] Response succeeded

I set the RUST_LOG=debug but not additional information.

**Comment by Rauks** [comment]
Hi,

I encountered the same issue, and made the same mistake than you @mark-e-hoffman 😄 

You added the prometheus fairing as a managed state (https://rocket.rs/v0.5-rc/guide/state/#managed-state) by doing:
```rust
.manage(prometheus.clone())
```

You should use `attach` to attach the fairing instead (https://rocket.rs/v0.5-rc/guide/fairings/#fairings):
```rust
.attach(prometheus.clone())
```

**Comment by sd2k** [comment]
Good catch @Rauks! Thanks for replying here, glad you found the cause. I'm not even sure there's a good way of avoiding that, unless I split the struct up into 2 somehow... I'll give it some thought.
